### PR TITLE
Add WOLFSSL_FORCE_AUTO_RETRY option: force retrying of network reads

### PR DESCRIPTION
**Pull request by julek-wolfssl**
Addresses issue #4593

### CODE REVIEW

File: src/internal.c [hunk 1]
```diff
@@ -65,6 +65,12 @@
  *     may be received by a client. To support detecting this, peek will
  *     return WOLFSSL_ERROR_WANT_READ.
  *     This define turns off this behaviour.
+ * WOLFSSL_FORCE_AUTO_RETRY
```
Comment (dgarske) [inline src/internal.c:68]: So defining this macro forces the network reads to behave as blocking? Is there a reason not to implement runtime support using the `SSL_MODE_AUTO_RETRY` option? Also can you check review the code comment at ssl.h:2136 for `SSL_MODE_AUTO_RETRY`? It seems wrong.

```
SSL_MODE_AUTO_RETRY = 3, /* wolfSSL default is to block with blocking io and auto retry */
```

File: src/internal.c [hunk 2]
```diff
@@ -65,6 +65,12 @@
...
+ *     This will force wolfSSL to always retry to read when it receives a
```
Comment (dgarske) [inline src/internal.c:69]: This option will made a wolfSSL read blocking.

**Review by dgarske — CHANGES_REQUESTED** [review]
Is there a specific open source project this fix is being made for?

File: src/internal.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/internal.c:68]: I wanted to keep this PR minimum but if you would prefer a runtime solution, I can add the necessary API to make that work. Still, I don't think that this is a good default for wolfSSL. If you want me to implement a runtime solution, I'll make the `SSL_MODE_AUTO_RETRY` option on by default only with `OPENSSL_COMPATIBLE_DEFAULTS`.

**Comment by julek-wolfssl** [comment]
@dgarske This is being made in response to an issue: https://github.com/wolfSSL/wolfssl/issues/4593. In another issue this user is asking in relation to the fetchmail project so I would guess this issue comes from work on the same project.

File: src/internal.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/internal.c:68]: I'll update the `SSL_MODE_AUTO_RETRY` comment to match the real behaviour.

**Comment by mandree** [comment]
> @dgarske This is being made in response to an issue: #4593. In another issue this user is asking in relation to the fetchmail project so I would guess this issue comes from work on the same project.

True, but fetchmail no longer requires it after recent updates to its TLS code.

**Comment by julek-wolfssl** [comment]
I changed this to set `SSL_MODE_AUTO_RETRY` with `OPENSSL_COMPATIBLE_DEFAULTS` and added `wolfSSL_CTX_clear_mode` to allow for clearing this option.

File: src/internal.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/internal.c]: Comment fixed.

**Review by dgarske — APPROVED** [review]

**Pull request merged**